matrix_sdk_crypto/src/store/memorystore.rs

```diff
@@ -30,9 +32,17 @@ use super::{
 };
 use crate::{
     identities::{ReadOnlyDevice, UserIdentities},
+    key_request::OutgoingKeyRequest,
     olm::{OutboundGroupSession, PrivateCrossSigningIdentity},
 };
 
+fn encode_key_info(info: &RequestedKeyInfo) -> String {
+    format!(
+        "{}{}{}{}",
+        &info.room_id, &info.sender_key, &info.algorithm, &info.session_id
```

Comment:
```suggestion
        info.room_id, info.sender_key, info.algorithm, info.session_id
```

`format!` never consumes its arguments, even if it looks like it should, so you can get of these `&`s.